fenced-ring-buffer/src/async_reader.rs

```diff
@@ -71,47 +74,47 @@ where
         // If any entries were missed and there is a stored prefix, then the entry after the prefix was missed.
         // The prefix is dropped and added to the missed count.
         // Note: the read sequence number is updated above because the prefix was included in the last read
-        if Into::<u64>::into(n_missed_before_read) > 0 && self.drop_prefix() {
+        if u64::from(n_missed_before_read) > 0 && self.drop_prefix() {
             n_missed_before_read += 1;
         }
 
         // Perform reads into snapshot buffer up to write sequence number
         let first_read_seqn = self.read_seqn;
-        let mut buf_snapshot: Vec<E> = Vec::new();
```

Comment:
It is probably wise / necessary to call `self.buf_snapshot.clear()` here because the `clear` call near the end of the function may not be called if there is an early-return (e.g. if an error is returned via a `?`).